sale_partner_order_template/models/sale_order.py

```diff
@@ -0,0 +1,13 @@
+# Copyright 2025 Binhex <https://www.binhex.cloud>
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, models
+
+
+class SaleOrder(models.Model):
+    _inherit = "sale.order"
+
+    @api.onchange("partner_id")
+    def onchange_partner_id_set_order_template(self):
```

Comment:
IMHO you should override the original compute method with a depends on `partner_id` 

At least onchange method should not be public